File: packages/shared/src/types/org-access.ts
export interface OrgAccessRequestBody {
  email: string;
}

export interface OrgAccessRequestResponse {
  verification_id: string;
  expires_in_seconds: number;
}

export interface OrgAccessConfirmBody {
  verification_id: string;
  code: string;
}

export interface OrgAccessConfirmResponse {
  org_verified_at: string;
  org_verified_domain: string;
}

File: services/api/app/models/org_access.py
from pydantic import BaseModel


class OrgAccessRequest(BaseModel):
    email: str


class OrgAccessRequestResponse(BaseModel):
    verification_id: str
    expires_in_seconds: int


class OrgAccessConfirm(BaseModel):
    verification_id: str
    code: str


class OrgAccessConfirmResponse(BaseModel):
    org_verified_at: str
    org_verified_domain: str
